source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1348,6 +1348,31 @@ ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::ClusterEntry(
                                                      parent.parent_.random_, cluster->lbConfig());
       break;
     }
+    case LoadBalancerType::LoadBalancingPolicyConfig: {
```

Comment:
This occurs off of the main thread, so this won't work since we need to verify the correctness of the configuration prior to when we get here. The way I would recommend doing this is processing the configuration on the main thread next to the other LBs, and storing a factory that we know will work, and then using that factory in the appropriate process in worker context.

Reply:
There already is a `lb_factory_`. I noticed this is used for the consistent LBs that are built on the main thread, I wonder if this pattern somehow can be reused.

Reply:
This commit (https://github.com/envoyproxy/envoy/pull/15827/commits/5e8ce8c0bff1941fbeceaf191e8a07a295b9ae91) attempts to mimic lb_factory_ more closely: getFactory is now pulled in to the main thread. I had a hard time coming up with a scenario where I could trigger a config error in the non-main threads, and I can't find where this verification comes from, so this is not much more than a shot in the dark.

I'll continue investigating this, and any extra pointers may speed me up!